target_chains/ethereum/contracts/contracts/pyth/Pyth.sol

```diff
@@ -469,15 +469,13 @@ abstract contract Pyth is
 
                         uint publishTime = uint(info.publishTime);
                         // Check the publish time of the price is within the given range
-                        // if it is not, then set the id to 0 to indicate that this price id
-                        // still does not have a valid price feed. This will allow other updates
-                        // for this price id to be processed.
+                        // and only fill the priceFeedsInfo if it is.
+                        // If is not, default id value of 0 will still be set and
+                        // this will allow other updates for this price id to be processed.
                         if (
-                            publishTime < minPublishTime ||
-                            publishTime > maxPublishTime
+                            publishTime >= minPublishTime &&
+                            publishTime <= maxPublishTime
```

Comment:
"you can't specify a zero-length interval" is wrong. We don't have zero-length on inclusive end too. 

I decided to have inclusive end interval because it's more intuitive with the minPublishTime and maxPublishTime.

Reply:
"you can't specify a zero-length interval" is wrong. We don't have zero-length on inclusive end too. 

I decided to have inclusive end interval because it's more intuitive with the minPublishTime and maxPublishTime.

Reply:
I think the typical convention is >= and <, and following that will be more intuitive to most people.

Reply:
idk :D maybe you are right and it's very personal. I agree that with array ranges (slices) `[start, end)` is the convention. but when there is min & max, I think `[min, max]` is more intuitive.